src/syntax/ctt/operator.sml

```diff
@@ -12,6 +12,7 @@ struct
     | SQUASH of Sort.t
     | ENSEMBLE of Sort.t * Sort.t
     | DFUN | FUN | LAM | AP
+    | DFUN_DOM | DFUN_COD
```

Comment:
These operators take a term like `dfun(A; [x].B[x])` and extract the (co)domain! It's extensional because `dfun` is canonical.